cypress/integration/crd.spec.js

```diff
@@ -0,0 +1,167 @@
+      cy.get('mat-row').should('have.length', 1);
+    });
+  });
+
+  it('go to crd details page', () => {
+    cy.get('kd-crd-list').within(() => {
+      cy.get('a')
+        .first()
+        .click();
+    });
+    cy.url().should(
+      'include',
+      '/#/customresourcedefinition/foos.samplecontroller.k8s.io?namespace=default',
+    );
+
+    // loads correctly
+    cy.get('kd-object-meta').within(() => {
+      cy.get('kd-property')
+        .first()
+        .contains('foos.samplecontroller.k8s.io');
+    });
+
+    // has resource information
+    cy.get('[data-cy=resource-information]').within(() => {
+      cy.get('kd-property').should('have.length', 3);
+
+      cy.get('kd-property')
+        .eq(0)
+        .contains('v1alpha1');
+      cy.get('kd-property')
+        .eq(1)
+        .contains('Namespaced');
+      cy.get('kd-property')
+        .eq(2)
+        .contains('samplecontroller.k8s.io');
+    });
+
+    // has accepted names
+    cy.get('[data-cy=accepted-names]').within(() => {
+      cy.get('kd-property').should('have.length', 4);
+
+      cy.get('kd-property')
+        .eq(0)
+        .contains('foos');
+      cy.get('kd-property')
+        .eq(1)
+        .contains('foo');
+      cy.get('kd-property')
+        .eq(2)
+        .contains('Foo');
+      cy.get('kd-property')
+        .eq(3)
+        .contains('FooList');
+    });
+
+    // has empty object section
+    cy.get('kd-crd-object-list').within(() => {
+      cy.get('[data-cy=zero-state]').should('be.visible');
+    });
+
+    // has one version
+    cy.get('kd-crd-versions-list').within(() => {
+      cy.get('mat-row').should('have.length', 1);
+
+      cy.get('mat-cell')
+        .first()
+        .contains('v1alpha1');
+    });
+  });
+
+  it('create a crd object', () => {
+    cy.fixture('example-foo.yaml').then(object => {
+      cy.createResource({content: object, namespace: 'default'});
+      cy.reload();
+    });
+  });
+
+  it('crd objects list should contain the new object', () => {
+    cy.get('kd-crd-object-list').within(() => {
+      cy.get('[data-cy=zero-state]').should('not.be.visible');
+
+      cy.get('mat-row').should('have.length', 1);
+      cy.get('mat-cell')
+        .first()
+        .contains('example-foo');
+    });
+  });
+
+  it('go to object detail page', () => {
+    cy.get('kd-crd-object-list').within(() => {
+      cy.get('a')
+        .first()
+        .click();
+    });
+
+    cy.url().should(
+      'include',
+      '/#/customresourcedefinition/foos.samplecontroller.k8s.io/default/example-foo?namespace=default',
+    );
+
+    cy.get('kd-object-meta').within(() => {
+      // Name
+      cy.get('kd-property')
+        .eq(0)
+        .contains('example-foo');
+
+      // Namespace
+      cy.get('kd-property')
+        .eq(1)
+        .contains('default');
+    });
+  });
+
+  it('delete crd object', () => {
+    cy.server();
+    cy.route('DELETE', '/api/v1/_raw/foos.samplecontroller.k8s.io/**').as('deleteObject');
```

Comment:
It's an API test, not UI. UI should offer functionality to delete it.

Reply:
It is deleted via the UI. This is only telling Cypress to properly wait for the XHR to complete. The actual deletion occurs in following lines.

This follows your suggestion to use `route` and `alias` for waiting.